pkg/relayer/session/session.go

```diff
@@ -225,8 +225,10 @@ func (rp *relayerSessionsManager) validateConfig() error {
 // waitForBlock blocks until the block at the given height (or greater) is
 // observed as having been committed.
 func (rs *relayerSessionsManager) waitForBlock(ctx context.Context, height int64) client.Block {
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
#PUC why we need to cancel

Reply:
Anywhere we `#Subscribe()` to an observable, we need to either cancel the associated context or close the associated publish channel. The former is preferable for consistency because we may not always have a reference to the publish channel. For example, an observable (and it's corresponding publish channel) could be constructed in another scope that only returns the observable itself. In this case the only way to "close" the observable is to cancel the context.

Reply:
@bryanchriswhite Shold the observable be able to close itself?